## Section 4: Statistical Testing

## Objectives

- Understand and apply appropriate statistical tests to analyze data and draw meaningful conclusions.

## Input Data

Dataset/cleaned/combined_ml.csv

___

In [1]:
import os
current_dir = os.getcwd()
current_dir

'/Users/danielledelouw/Documents/code_institute/Pokemon_catch_them_all/Pokemon_Catch_Them_All/jupyter_notebooks'

In [2]:
os.chdir(os.path.dirname(current_dir))
print("You set a new current directory")

You set a new current directory


In [3]:
current_dir = os.getcwd()
current_dir

'/Users/danielledelouw/Documents/code_institute/Pokemon_catch_them_all/Pokemon_Catch_Them_All'

## Load Python Libraries

In [4]:
import pandas as pd
from scipy import stats





## Load Data

In [5]:
# Load the dataset
df = pd.read_csv('Dataset/cleaned/combined_ml.csv')
df.head()

First_pokemon  Second_pokemon  Winner  id_x First_Type_1  First_Total  \
0            266             298     298   266         Rock          300   
1            702             701     701   702        Grass          580   
2            191             668     668   191        Fairy          405   
3            237             683     683   237         Fire          250   
4            151             231     151   151         Rock          495   

   First_Hit_points  First_Attack  First_Defense  First_Special_attack  ...  \
0                50            64             50                    45  ...   
1                91            90             72                    90  ...   
2                55            40             85                    80  ...   
3                40            40             40                    70  ...   
4                70            60            125                   115  ...   

   id_y  Second_Type_1  Second_Total Second_Hit_points  Second_Attack  \
0   298          Grass           340                70             70   
1   701           Rock           580                91            129   
2   668        Psychic           485                75             75   
3   683         Dragon           485                77            120   
4   231            Bug           505                20             10   

   Second_Defense  Second_Special_attack  Second_Special_defense  \
0              40                     60                      40   
1              90                     72                      90   
2              75                    125                      95   
3              90                     60                      90   
4             230                     10                     230   

   Second_Speed  target  
0            60       0  
1           108       0  
2            40       0  
3            48       0  
4             5       1  

[5 rows x 22 columns]

In [ ]:
# 1. Paired t-test: First_Total vs Second_Total - Is there a significant difference in total stats between the first and second Pokémon?
t_stat, t_pval = stats.ttest_rel(df['First_Total'], df['Second_Total'])
print(" Paired t-test (First_Total vs Second_Total):")
print(f"  t-statistic = {t_stat:.4f}")
print(f"  p-value     = {t_pval:.4f}")
if t_pval < 0.05:
    print("There is a significant difference in total stats between the first and second Pokémon.")
else:
    print(" There is no significant difference in total stats between the first and second Pokémon.")
print()



 Paired t-test (First_Total vs Second_Total):
  t-statistic = 0.4216
  p-value     = 0.6733
 There is no significant difference in total stats between the first and second Pokémon.



In [8]:
# 2. Chi-Squared Test: First_Type_1 vs Target
# Create a contingency table of First_Type_1 vs target
chi_table = pd.crosstab(df['First_Type_1'], df['target'])

# Perform the Chi-Squared test
chi2_stat, chi2_pval, chi2_dof, chi2_expected = stats.chi2_contingency(chi_table)

# Print the results
print("Chi-Squared Test (First_Type_1 vs Target):")
print(f"  Chi² statistic     = {chi2_stat:.4f}")
print(f"  p-value            = {chi2_pval:.4f}")
print(f"  Degrees of freedom = {chi2_dof}")

# Interpret the result
if chi2_pval < 0.05:
    print("Pokémon type is significantly associated with the battle outcome.")
else:
    print("Pokémon type is not significantly associated with the battle outcome.")
print()

Chi-Squared Test (First_Type_1 vs Target):
  Chi² statistic     = 1230.0036
  p-value            = 0.0000
  Degrees of freedom = 17
Pokémon type is significantly associated with the battle outcome.



In [10]:
# 3. Pearson Correlation: First_Speed vs Target ===
corr, corr_pval = stats.pearsonr(df['First_Speed'], df['target'])
print("Pearson Correlation (First_Speed vs Target):")
print(f"  Correlation coefficient = {corr:.4f}")
print(f"  p-value                 = {corr_pval:.4f}")
if corr_pval < 0.05:
    print("There is a significant correlation between speed and winning.")
else:
    print("There is no significant correlation between speed and winning.")
print()

Pearson Correlation (First_Speed vs Target):
  Correlation coefficient = 0.4907
  p-value                 = 0.0000
There is a significant correlation between speed and winning.



In [11]:
# 4. Mann-Whitney U Test: First_Attack (Winners vs Losers) ===
winners = df[df['Winner'] == df['First_pokemon']]['First_Attack']
losers = df[df['Winner'] != df['First_pokemon']]['First_Attack']
mw_stat, mw_pval = stats.mannwhitneyu(winners, losers, alternative='two-sided')
print("Mann-Whitney U Test (First_Attack - Winners vs Losers):")
print(f"  U statistic = {mw_stat:.4f}")
print(f"  p-value     = {mw_pval:.4f}")
if mw_pval < 0.05:
    print("The attack values of winners and losers are significantly different.")
else:
    print("There is no significant difference in attack values between winners and losers.")

Mann-Whitney U Test (First_Attack - Winners vs Losers):
  U statistic = 374506418.5000
  p-value     = 0.0000
The attack values of winners and losers are significantly different.


### Summary of Statistical Test Results

- **Paired t-test (First_Total vs Second_Total):**
    - *t-statistic*: 0.4216
    - *p-value*: 0.6733  
    There is **no significant difference** in total stats between the first and second Pokémon.

- **Chi-Squared Test (First_Type_1 vs Target):**
    - *Chi² statistic*: 1230.0036
    - *Degrees of freedom*: 17
    - *p-value*: 4.82e-251  
    Pokémon type is **significantly associated** with the battle outcome.

- **Pearson Correlation (First_Speed vs Target):**
    - *Correlation coefficient*: 0.4907
    - *p-value*: 0.0  
    There is a **significant positive correlation** between speed and winning.

- **Mann-Whitney U Test (First_Attack - Winners vs Losers):**
    - *U statistic*: 374506418.5
    - *p-value*: 0.0  
    The attack values of winners and losers are **significantly different**.